requirements.txt

```diff
@@ -61,6 +61,7 @@ flake8-noqa==1.3.0
 flake8-comprehensions==3.10.1
 isort==5.11.4
 mypy==1.0.1
+pre-commit==3.1.1
```

Comment:
You are adding this, but I do not see that it is installed as a hook ??

Reply:
There is no way for a git server to distribute/enforce client-side hooks. (It would be a security issue).  Therefore, they must be [installed](https://pre-commit.com/#install) manually.  

In this case, with `pre-commit install`.

This PR does nothing to the CI, nor to developers that don't install pre-commit.

Reply:
it should and can be done in setup.py

Reply:
Package installation should not have any idea about the existence of repo-level QA tooling.  QA tooling should be installed externally from `setup.py`.

Imo the CI workflow should invoke `pre-commit install-hooks` and then cache `~/.cache/pre-commit`.  Or you can use the `pre-commit` GitHub Action to automate this

Reply:
That is a fair argument, but since it needs to be installed manually, the very least to do is to update README.rst to reflect that.